tinynn/graph/quantization/quantizable/gru.py

```diff
@@ -0,0 +1,341 @@
+        else:
+            hx = hidden
+            rh, zh, nh = self.hgates(hx).chunk(3, -1)
+            rgate = self.act1(self.add1.add(ri, rh))
+            zgate = self.act1(self.add2.add(zi, zh))
+            ngate = self.act2(self.add3.add(ni, self.mul1.mul(rgate, nh)))
+            hidden = self.add4.add(self.mul2.mul(self.sub1.add_scalar(torch.tensor(1), -zgate), ngate),
+                                   self.mul3.mul(zgate, hx))
+            result.append(hidden)
+            result_tensor = torch.stack(result, 0)
+            return result_tensor
+
+    def initialize_hidden(self, batch_size: int, is_quantized: bool = False) -> Tensor:
+        h = torch.zeros(batch_size, self.hidden_size)
+        if is_quantized:
+            (h_scale, h_zp) = self.initial_hidden_state_qparams
+            h = torch.quantize_per_tensor(h, scale=h_scale, zero_point=h_zp, dtype=self.hidden_state_dtype)
+        return h
+
+    def _get_name(self):
+        return 'QuantizableGRUCell'
+
+    @classmethod
+    def from_params(cls, wi, wh, bi=None, bh=None):
+        """Uses the weights and biases to create a new GRU cell.
+        Args:
+            wi, wh: Weights for the input and hidden layers
+            bi, bh: Biases for the input and hidden layers
+        """
+        assert (bi is None) == (bh is None)  # Either both None or both have values
+        input_size = wi.shape[1]
+        hidden_size = wh.shape[1]
+        cell = cls(input_dim=input_size, hidden_dim=hidden_size,
+                   bias=(bi is not None))
+        cell.igates.weight = torch.nn.Parameter(wi)
+        if bi is not None:
+            cell.igates.bias = torch.nn.Parameter(bi)
+        cell.hgates.weight = torch.nn.Parameter(wh)
+        if bh is not None:
+            cell.hgates.bias = torch.nn.Parameter(bh)
+        return cell
+
+    @classmethod
+    def from_float(cls, other):
+        assert type(other) == cls._FLOAT_MODULE
+        assert hasattr(other, 'qconfig'), "The float module must have 'qconfig'"
+        observed = cls.from_params(other.weight_ih, other.weight_hh,
+                                   other.bias_ih, other.bias_hh)
+        observed.qconfig = other.qconfig
+        observed.igates.qconfig = other.qconfig
+        observed.hgates.qconfig = other.qconfig
+        return observed
+
+
+class _GRUSingleLayer(torch.nn.Module):
+    r"""A single one-directional GRU layer.
+    The difference between a layer and a cell is that the layer can process a
+    sequence, while the cell only expects an instantaneous value.
+    """
+    def __init__(self, input_dim: int, hidden_dim: int, bias: bool = True,
+                 device=None, dtype=None) -> None:
+        factory_kwargs = {'device': device, 'dtype': dtype}
+        super().__init__()
+        self.cell = GRUCell(input_dim, hidden_dim, bias=bias, **factory_kwargs)
+
+    def forward(self, x: Tensor, hidden: Optional[Tensor] = None):
+        # result = []
```

Comment:
Please remove the unused code block.

Reply:
Alternatively, if you want to preserve the code, please introduce an option to switch the codepath.